### PR TITLE
Cu 57wk7q clark chen

### PR DESCRIPTION
**Pull request by Clark-spec**
Create entity
CRUD for lotteryPool

### CODE REVIEW

File: lottery-clientapi/src/main/java/com/arbfintech/station/lottery/clientapi/controller/api/LotteryPoolApiController.java [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+
+    @PostMapping("/create")
+    public String createNewLotteryPool(
+            @RequestParam Long lotteryPoolSize) {
+        String newLotteryPool = lotteryPoolService.createLotteryPool(lotteryPoolSize);
+        return newLotteryPool;
+    }
+
+    @PostMapping("/addOwner")
+    public String addLotteryPoolOwner(
```
Comment (roushanyyzz) [inline lottery-clientapi/src/main/java/com/arbfintech/station/lottery/clientapi/controller/api/LotteryPoolApiController.java:32]: add-owner

File: lottery-clientapi/src/main/java/com/arbfintech/station/lottery/clientapi/controller/api/LotteryPoolApiController.java [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+
+    private static Logger LOGGER = LoggerFactory.getLogger(LotteryPoolApiController.class);
+
+    @Autowired
+    private LotteryPoolService lotteryPoolService;
+
+    @PostMapping("/create")
+    public String createNewLotteryPool(
+            @RequestParam Long lotteryPoolSize) {
+        String newLotteryPool = lotteryPoolService.createLotteryPool(lotteryPoolSize);
```
Comment (roushanyyzz) [inline lottery-clientapi/src/main/java/com/arbfintech/station/lottery/clientapi/controller/api/LotteryPoolApiController.java:27]: createNewLotteryPool(
            @RequestParam Long lotteryPoolSize
)

File: lottery-clientapi/src/main/java/com/arbfintech/station/lottery/clientapi/controller/api/LotteryPoolApiController.java [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+    @DeleteMapping("/delete")
+    public String deleteSelect(
+            @RequestParam Long lotteryHolderId
+    ){
+        LOGGER.info("Start to delete lottery that have already been selected");
+        return lotteryPoolService.deleteSelect(lotteryHolderId);
+    }
+
+    @GetMapping("/querySelected")
+    public String querySelected(){
```
Comment (roushanyyzz) [inline lottery-clientapi/src/main/java/com/arbfintech/station/lottery/clientapi/controller/api/LotteryPoolApiController.java:49]: Do not use hump names in urls

File: lottery-clientapi/src/main/java/com/arbfintech/station/lottery/clientapi/controller/api/LotteryPoolApiController.java [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+        Integer lotteryPoolStatus = LotteryPoolStatusEnum.ENABLED.getValue();
+        LOGGER.info("Start looking for selected lists");
+        return lotteryPoolService.findByStatus(lotteryPoolStatus);
+    }
+
+    @GetMapping("/queryNotSelected")
+    public String queryNotSelected(){
```
Comment (roushanyyzz) [inline lottery-clientapi/src/main/java/com/arbfintech/station/lottery/clientapi/controller/api/LotteryPoolApiController.java:56]: Do not use hump names in urls

File: lottery-clientapi/src/main/java/com/arbfintech/station/lottery/clientapi/controller/api/LotteryPoolApiController.java [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+        Integer lotteryPoolStatus = LotteryPoolStatusEnum.INITED.getValue();
+        LOGGER.info("Start looking for unselected lists");
+        return lotteryPoolService.findByStatus(lotteryPoolStatus);
+    }
+
+    @GetMapping("/queryAll")
+    public String queryAll(){
```
Comment (roushanyyzz) [inline lottery-clientapi/src/main/java/com/arbfintech/station/lottery/clientapi/controller/api/LotteryPoolApiController.java:63]: Do not use hump names in urls

File: lottery-clientapi/src/main/java/com/arbfintech/station/lottery/clientapi/service/LotteryHolderService.java [hunk 6]
```diff
@@ -217,7 +210,9 @@ public String updateLotteryHolderPassword(String loginDataStr){
         password = password.toUpperCase();
         LOGGER.warn("Start to update password , customerId:{}, new password:{}", lotteryHolder.getId(), password);
         String encryptPassword = CryptUtil.md5(password + lotteryHolder.getSalt());
-        lotteryHolder.setStatus(1);
+        lotteryHolder.setStatus(LotteryHolderStatusEnum.ENABLED.getValue());
+        System.out.println(LotteryHolderStatusEnum.ENABLED.getValue());
```
Comment (roushanyyzz) [inline lottery-clientapi/src/main/java/com/arbfintech/station/lottery/clientapi/service/LotteryHolderService.java:214]: Don't print

File: lottery-clientapi/src/main/java/com/arbfintech/station/lottery/clientapi/service/LotteryPoolService.java [hunk 7]
```diff
@@ -0,0 +1,120 @@
...
+    private static final Logger LOGGER = LoggerFactory.getLogger(LotteryPool.class);
+
+    @Autowired
+    private LotteryPoolRepository lotteryPoolRepository;
+
+    @Autowired
+    private LotteryHolderService lotteryHolderService;
+
+    public String createLotteryPool(Long lotteryPoolSize){
+        LOGGER.info("Start to create new lottery pool, size{}",lotteryPoolSize);
```
Comment (roushanyyzz) [inline lottery-clientapi/src/main/java/com/arbfintech/station/lottery/clientapi/service/LotteryPoolService.java:34]: lotteryPoolSize use Int

**Review by roushanyyzz — CHANGES_REQUESTED** [review]
Update

File: lottery-clientapi/src/main/java/com/arbfintech/station/lottery/clientapi/controller/api/LotteryPoolApiController.java (hunk 1 above)
Comment (Clark-spec) [reply lottery-clientapi/src/main/java/com/arbfintech/station/lottery/clientapi/controller/api/LotteryPoolApiController.java:32]: update

File: lottery-clientapi/src/main/java/com/arbfintech/station/lottery/clientapi/controller/api/LotteryPoolApiController.java (hunk 2 above)
Comment (Clark-spec) [reply lottery-clientapi/src/main/java/com/arbfintech/station/lottery/clientapi/controller/api/LotteryPoolApiController.java:27]: update

File: lottery-clientapi/src/main/java/com/arbfintech/station/lottery/clientapi/controller/api/LotteryPoolApiController.java (hunk 3 above)
Comment (Clark-spec) [reply lottery-clientapi/src/main/java/com/arbfintech/station/lottery/clientapi/controller/api/LotteryPoolApiController.java:49]: update

File: lottery-clientapi/src/main/java/com/arbfintech/station/lottery/clientapi/controller/api/LotteryPoolApiController.java (hunk 4 above)
Comment (Clark-spec) [reply lottery-clientapi/src/main/java/com/arbfintech/station/lottery/clientapi/controller/api/LotteryPoolApiController.java:56]: update

File: lottery-clientapi/src/main/java/com/arbfintech/station/lottery/clientapi/controller/api/LotteryPoolApiController.java (hunk 5 above)
Comment (Clark-spec) [reply lottery-clientapi/src/main/java/com/arbfintech/station/lottery/clientapi/controller/api/LotteryPoolApiController.java:63]: update

File: lottery-clientapi/src/main/java/com/arbfintech/station/lottery/clientapi/service/LotteryHolderService.java (hunk 6 above)
Comment (Clark-spec) [reply lottery-clientapi/src/main/java/com/arbfintech/station/lottery/clientapi/service/LotteryHolderService.java]: update

File: lottery-clientapi/src/main/java/com/arbfintech/station/lottery/clientapi/service/LotteryPoolService.java (hunk 7 above)
Comment (Clark-spec) [reply lottery-clientapi/src/main/java/com/arbfintech/station/lottery/clientapi/service/LotteryPoolService.java:37]: update

File: lottery-clientapi/src/main/java/com/arbfintech/station/lottery/clientapi/controller/api/LotteryPoolApiController.java [hunk 8]
```diff
@@ -0,0 +1,84 @@
...
+
+    private static Logger LOGGER = LoggerFactory.getLogger(LotteryPoolApiController.class);
+
+    @Autowired
+    private LotteryPoolService lotteryPoolService;
+
+    @PostMapping("/create-lottery-pool")
+    public String createLotteryPool(
+            @RequestParam int lotteryPoolSize) {
+        String newLotteryPool = lotteryPoolService.createLotteryPool(lotteryPoolSize);
```
Comment (roushanyyzz) [inline lottery-clientapi/src/main/java/com/arbfintech/station/lottery/clientapi/controller/api/LotteryPoolApiController.java:27]: public String createLotteryPool(
            @RequestParam int lotteryPoolSize
) {

Comment (Clark-spec) [reply lottery-clientapi/src/main/java/com/arbfintech/station/lottery/clientapi/controller/api/LotteryPoolApiController.java:28]: update

File: lottery-clientapi/src/main/java/com/arbfintech/station/lottery/clientapi/service/LotteryPoolService.java [hunk 9]
```diff
@@ -0,0 +1,161 @@
...
+import java.util.stream.Collectors;
+
+/**
+ * @author Clark
+ * @date 2020年11月23日13:50:51
+ */
+@Service
+public class LotteryPoolService {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(LotteryPool.class);
```
Comment (roushanyyzz) [inline lottery-clientapi/src/main/java/com/arbfintech/station/lottery/clientapi/service/LotteryPoolService.java:28]: LotteryPool -> LotteryPoolService

File: lottery-clientapi/src/main/java/com/arbfintech/station/lottery/clientapi/service/LotteryPoolService.java [hunk 10]
```diff
@@ -0,0 +1,161 @@
...
+
+    @Autowired
+    private LotteryHolderService lotteryHolderService;
+
+    public String createLotteryPool(int lotteryPoolSize){
+        LOGGER.info("Start to create new lottery pool, size{}",lotteryPoolSize);
+        for (int i = 1; i <= lotteryPoolSize; i++){
+            LotteryPool lotteryPool = new LotteryPool();
+            lotteryPool.setNumber(i);
+            lotteryPool.setStatus(0);
```
Comment (roushanyyzz) [inline lottery-clientapi/src/main/java/com/arbfintech/station/lottery/clientapi/service/LotteryPoolService.java:41]: use enum

File: lottery-clientapi/src/main/java/com/arbfintech/station/lottery/clientapi/service/LotteryPoolService.java [hunk 11]
```diff
@@ -0,0 +1,161 @@
...
+            lotteryPoolRepository.save(lotteryPool);
+        }
+        return AjaxResult.success("Create new lottery success");
+    }
+
+    public LotteryPool findByLotteryPoolNum(Long lotteryPoolNum){
```
Comment (roushanyyzz) [inline lottery-clientapi/src/main/java/com/arbfintech/station/lottery/clientapi/service/LotteryPoolService.java:47]: Why number is long?

File: lottery-clientapi/src/main/java/com/arbfintech/station/lottery/clientapi/service/LotteryPoolService.java [hunk 12]
```diff
@@ -0,0 +1,161 @@
...
+        LOGGER.info("Find all lottery pool data");
+        List<LotteryPool> lotteryPoolList = lotteryPoolRepository.findAll();
+        JSONArray lotteryPoolListJSON = JSON.parseArray(JSON.toJSONString(lotteryPoolList));
+        return AjaxResult.success(lotteryPoolListJSON);
+    }
+
+    public String findByStatus(Integer lotteryPoolStatus){
+        LOGGER.info("Query by lottery pool status,lotteryPoolStatus{}",lotteryPoolStatus);
+        List<LotteryPool> lotteryPoolList = lotteryPoolRepository.findByStatusEquals(lotteryPoolStatus);
+        JSONArray lotteryPoolArray = JSON.parseArray(JSON.toJSONString(lotteryPoolList));
```
Comment (roushanyyzz) [inline lottery-clientapi/src/main/java/com/arbfintech/station/lottery/clientapi/service/LotteryPoolService.java:64]: Don't use JSONArray lotteryPoolArray = JSON.parseArray(JSON.toJSONString(lotteryPoolList));
 use : return AjaxResult.success(lotteryPoolList);

File: lottery-clientapi/src/main/java/com/arbfintech/station/lottery/clientapi/service/LotteryPoolService.java [hunk 13]
```diff
@@ -0,0 +1,161 @@
...
+        return AjaxResult.success(lotteryPoolArray);
+    }
+
+    public String addOwnerAndStatusByNum(Long lotteryHolderId,Long Num){
+        LotteryPool lotteryPool = findByLotteryPoolNum(Num);
+        if (Objects.isNull(lotteryPool)){
+            LOGGER.warn("Num is not exit,Num{}", Num);
```
Comment (roushanyyzz) [inline lottery-clientapi/src/main/java/com/arbfintech/station/lottery/clientapi/service/LotteryPoolService.java:71]: Num{} add space

File: lottery-clientapi/src/main/java/com/arbfintech/station/lottery/clientapi/service/LotteryPoolService.java [hunk 14]
```diff
@@ -0,0 +1,161 @@
...
+            return AjaxResult.failure();
+        }
+        if (LotteryPoolStatusEnum.DISABLED.getValue().equals(lotteryPool.getStatus())){
+            LOGGER.warn("The time has expired, can not choose");
+            return AjaxResult.failure();
+        }
+        if ( LotteryPoolStatusEnum.ENABLED.getValue().equals(lotteryPool.getStatus())){
```
Comment (roushanyyzz) [inline lottery-clientapi/src/main/java/com/arbfintech/station/lottery/clientapi/service/LotteryPoolService.java:78]: Status just enable and disabled? Has something else?

File: lottery-clientapi/src/main/java/com/arbfintech/station/lottery/clientapi/service/LotteryPoolService.java [hunk 15]
```diff
@@ -0,0 +1,161 @@
...
+            LOGGER.warn("This num has been selected,Num{}", Num);
```
Comment (roushanyyzz) [inline lottery-clientapi/src/main/java/com/arbfintech/station/lottery/clientapi/service/LotteryPoolService.java:79]: Space

Comment (Clark-spec) [reply lottery-clientapi/src/main/java/com/arbfintech/station/lottery/clientapi/service/LotteryPoolService.java:79]: update

File: lottery-clientapi/src/main/java/com/arbfintech/station/lottery/clientapi/service/LotteryPoolService.java (hunk 14 above)
Comment (Clark-spec) [reply lottery-clientapi/src/main/java/com/arbfintech/station/lottery/clientapi/service/LotteryPoolService.java:78]: Status has three.Enable is selected.Disable means no longer effective.Inited is selectable.

File: lottery-clientapi/src/main/java/com/arbfintech/station/lottery/clientapi/service/LotteryPoolService.java (hunk 13 above)
Comment (Clark-spec) [reply lottery-clientapi/src/main/java/com/arbfintech/station/lottery/clientapi/service/LotteryPoolService.java:71]: update

File: lottery-clientapi/src/main/java/com/arbfintech/station/lottery/clientapi/service/LotteryPoolService.java (hunk 12 above)
Comment (Clark-spec) [reply lottery-clientapi/src/main/java/com/arbfintech/station/lottery/clientapi/service/LotteryPoolService.java:64]: update

File: lottery-clientapi/src/main/java/com/arbfintech/station/lottery/clientapi/service/LotteryPoolService.java (hunk 11 above)
Comment (Clark-spec) [reply lottery-clientapi/src/main/java/com/arbfintech/station/lottery/clientapi/service/LotteryPoolService.java:47]: update

File: lottery-clientapi/src/main/java/com/arbfintech/station/lottery/clientapi/service/LotteryPoolService.java (hunk 10 above)
Comment (Clark-spec) [reply lottery-clientapi/src/main/java/com/arbfintech/station/lottery/clientapi/service/LotteryPoolService.java:41]: update

File: lottery-clientapi/src/main/java/com/arbfintech/station/lottery/clientapi/service/LotteryPoolService.java (hunk 9 above)
Comment (Clark-spec) [reply lottery-clientapi/src/main/java/com/arbfintech/station/lottery/clientapi/service/LotteryPoolService.java:28]: update

**Comment by Clark-spec** [comment]
Show the number

**Comment by Clark-spec** [comment]
Show the number

**Comment by Clark-spec** [comment]
update

**Comment by Clark-spec** [comment]
update

**Comment by Clark-spec** [comment]
update

**Pull request closed without merging**